feature/bgp/gracefulrestart/otg_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go

```diff
@@ -0,0 +1,677 @@
+	gnoiClient := dut.RawAPIs().GNOI(t)
+	killRequest := &gnps.KillProcessRequest{Name: pName, Pid: pID, Signal: gnps.KillProcessRequest_SIGNAL_TERM, Restart: true}
+	killResponse, err := gnoiClient.System().KillProcess(context.Background(), killRequest)
+	t.Logf("Got kill process response: %v\n\n", killResponse)
+	if err != nil {
+		t.Fatalf("Failed to execute gNOI Kill Process, error received: %v", err)
+	}
+}
+
+func TestTrafficWithGracefulRestartReceiver(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+
+	// Configure interface on the DUT
+	t.Log("Start DUT interface Config")
+	configureDUT(t, dut)
+
+	// Configure BGP+Neighbors on the DUT
+	t.Log("Configure BGP with Graceful Restart option under Global Bgp")
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	nbrList := buildNbrList(ateAS)
+	dutConf := bgpWithNbr(dutAS, nbrList, dut)
+	gnmi.Replace(t, dut, dutConfPath.Config(), dutConf)
+	fptest.LogQuery(t, "DUT BGP Config", dutConfPath.Config(), gnmi.GetConfig(t, dut, dutConfPath.Config()))
+
+	// ATE Configuration.
+	t.Log("Start ATE Config")
+	config := configureATE(t, ate)
+
+	// Verify Port Status
+	t.Run("verifyDUTPorts", func(t *testing.T) {
+		t.Log("Verifying port status")
+		verifyPortsUp(t, dut.Device)
+	})
+	t.Run("VerifyBGPParameters", func(t *testing.T) {
+		t.Log("Check BGP parameters")
+		checkBgpStatus(t, dut)
+	})
+
+	var startTime time.Time
+	// Send Graceful Restart Trigger from ATE to DUT within the GR timer configured on the DUT
+	t.Run("InitiateGracefulRestartWithinDUTGRTimer", func(t *testing.T) {
+		// starting traffic before graceful restart
+		t.Log("Starting traffic before Graceful restrat Trigger from OTG")
+		ate.OTG().StartTraffic(t)
+		startTime = time.Now()
+		t.Log("Send Traffic while GR timer counting down. Traffic should pass as BGP GR is enabled!")
+		t.Log("Send Graceful Restart Trigger from OTG to DUT -- trigger GR Restart timer(60s) < configured GR restart Timer(120s)")
+		ate.OTG().SetControlAction(t, createGracefulRestartAction(t, []string{"atedst.BGP4.peer"}, 60))
+	})
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(ateDst.IPv4)
+
+	t.Run("VerifyBGPNOTEstablished", func(t *testing.T) {
+		t.Log("Waiting for BGP neighbor to go to ACTIVE state after GR Trigger to DUT...")
+		bgpState := oc.Bgp_Neighbor_SessionState_ACTIVE
+		_, ok := gnmi.Watch(t, dut, nbrPath.SessionState().State(), 2*time.Minute, func(val *ygnmi.Value[oc.E_Bgp_Neighbor_SessionState]) bool {
+			currState, ok := val.Val()
+			return ok && currState == bgpState
+		}).Await(t)
+		if !ok {
+			fptest.LogQuery(t, "BGP reported state", nbrPath.State(), gnmi.Get(t, dut, nbrPath.State()))
+			t.Errorf("BGP session did not go Down as expected")
+		}
+	})
+
+	const stopDuration = 45 * time.Second
```

Comment:
Should this either be a constant or based on the 60 second value (as a constant?)

Reply:
Done